Packs/Base/Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -8637,11 +8838,21 @@ def update_integration_context(context, object_keys=None, sync=True):
 
 
 class DemistoException(Exception):
-    def __init__(self, message, exception=None, res=None, *args):
+    def __init__(self, message, exception=None, res=None, error_type=None, *args):
         self.res = res
         self.message = message
         self.exception = exception
-        super(DemistoException, self).__init__(message, exception, *args)
+        self.error_type = error_type
+        super(DemistoException, self).__init__(message, exception, error_type, *args)
+
+        if self.error_type:
+            # If an error is passed to DemistoException, then check if metrics reporting is supported. If so,
+            # instantiate an ExecutionMetrics instance and report the metrics
+            execution_metrics = ExecutionMetrics()
+            if execution_metrics.is_supported():
+                execution_metrics.update_metrics(metric_type=error_type, metric_value=1)
+                exc_metrics = execution_metrics.metrics
+                return_results(CommandResults(execution_metrics=exc_metrics))
```

Comment:
We should not return_results in DemistoException init code.
This code should happen in the integration code.